crates/cli-support/src/js/binding.rs

```diff
@@ -548,13 +548,16 @@ fn instruction(js: &mut JsBuilder, instr: &Instruction, log_error: &mut bool) ->
         | Instruction::CallExport(_)
         | Instruction::CallAdapter(_)
         | Instruction::CallTableElement(_)
-        | Instruction::DeferCallCore(_) => {
+        | Instruction::DeferFree { .. } => {
             let invoc = Invocation::from(instr, js.cx.module)?;
-            let (params, results) = invoc.params_results(js.cx);
-
+            let (mut params, results) = invoc.params_results(js.cx);
             let mut args = Vec::new();
             let tmp = js.tmp();
             if invoc.defer() {
+                if let Instruction::DeferFree { .. } = instr {
```

Comment:
Right now the only instruction that has `defer` is `DeferFree`, but I added this check nonetheless, I hope that makes sense.